src/d3d8/d3d8_device.h

```diff
@@ -151,14 +151,15 @@ namespace dxvk {
             UINT iBackBuffer,
             D3DBACKBUFFER_TYPE Type,
             IDirect3DSurface8** ppBackBuffer) {
+      InitReturnPtr(ppBackBuffer);
       
       if (iBackBuffer >= m_backBuffers.size() || m_backBuffers[iBackBuffer] == nullptr) {
         Com<d3d9::IDirect3DSurface9> pSurface9;
         HRESULT res = GetD3D9()->GetBackBuffer(0, iBackBuffer, (d3d9::D3DBACKBUFFER_TYPE)Type, &pSurface9);
 
         if (FAILED(res)) return res;
         
-        m_backBuffers[iBackBuffer] = ref(new D3D8Surface(this, std::move(pSurface9)));
+        m_backBuffers[iBackBuffer] = new D3D8Surface(this, std::move(pSurface9));
         *ppBackBuffer = m_backBuffers[iBackBuffer].ref();
```

Comment:
Not 100% on this.

Reply:
As I've stated in point 1) above privateComObject.ref() already increments the public ref on an object. We were doing it twice here.